libs/vkd3d/command.c

```diff
@@ -6313,29 +6374,22 @@ static void STDMETHODCALLTYPE d3d12_command_list_ResolveQueryData(d3d12_command_
             iface, heap, type, start_index, query_count,
             dst_buffer, aligned_dst_buffer_offset);
 
-    d3d12_command_list_track_query_heap(list, query_heap);
-
-    /* Vulkan is less strict than D3D12 here. Vulkan implementations are free
-     * to return any non-zero result for binary occlusion with at least one
-     * sample passing, while D3D12 guarantees that the result is 1 then.
-     *
-     * For example, the Nvidia binary blob drivers on Linux seem to always
-     * count precisely, even when it was signalled that non-precise is enough.
-     */
-    if (type == D3D12_QUERY_TYPE_BINARY_OCCLUSION)
-        FIXME_ONCE("D3D12 guarantees binary occlusion queries result in only 0 and 1.\n");
-
     if (!d3d12_resource_is_buffer(buffer))
     {
         WARN("Destination resource is not a buffer.\n");
         return;
     }
 
+    d3d12_command_list_track_query_heap(list, query_heap);
     d3d12_command_list_end_current_render_pass(list, true);
 
     VK_CALL(vkCmdCopyQueryPoolResults(list->vk_command_buffer, query_heap->vk_query_pool,
             start_index, query_count, buffer->vk_buffer, buffer->heap_offset + aligned_dst_buffer_offset,
             get_query_stride(type), VK_QUERY_RESULT_64_BIT | VK_QUERY_RESULT_WAIT_BIT));
+
+    if (type == D3D12_QUERY_TYPE_BINARY_OCCLUSION)
```

Comment:
Do we expect applications to not batch here? Resolving one query at a time sounds like it would get "fun".

Reply:
Apps should batch if they aren't completely braindead. The D3D12 API kind of implies that this is implemented as some compute shader dispatch.

Reply:
Ok.